pyhive/hive.py

```diff
@@ -85,11 +88,16 @@ def __init__(self, host, port=10000, username=None, database='default', auth='NO
         username = username or getpass.getuser()
         configuration = configuration or {}
 
-        if (password is not None) != (auth == 'LDAP'):
-            raise ValueError("Password should be set if and only if in LDAP mode; "
-                             "Remove password or add auth='LDAP'")
+        if (password is not None) != (auth == 'LDAP' or mode == 'HTTP'):
+            raise ValueError("password should be set if and only if in LDAP mode or HTTP mode")
         if (kerberos_service_name is not None) != (auth == 'KERBEROS'):
             raise ValueError("kerberos_service_name should be set if and only if in KERBEROS mode")
+        if (http_path is not None) and (mode != 'HTTP'):
+            raise ValueError("http_path should be set if and only if in HTTP mode")
```

Comment:
Message is slightly wrong. Should be "http_path may be set only if in HTTP mode", since it's not mandatory to override in HTTP mode.